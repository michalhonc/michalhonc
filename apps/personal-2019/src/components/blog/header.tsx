import * as React from "react";
import { useIntl, Link, changeLocale } from "gatsby-plugin-intl";

import { Logo } from "../logo";
import { Nav } from "../nav";

import "../../styles/blog.scss";

interface IProps {
  canChangeLocal?: boolean;
}

export const Header: React.FC<IProps> = ({ canChangeLocal = true }) => {
  const intl = useIntl();

  return (
    <div className="Blog-header">
      <Logo />

      <Nav canChangeLocal={canChangeLocal}>
        <li>
          <Link to="/">{intl.formatMessage({ id: "aboutMe" })}</Link>
        </li>
        <li>
          <Link to="/blog">{intl.formatMessage({ id: "blog" })}</Link>
        </li>
        <li>
          <Link to="/#kontakty">{intl.formatMessage({ id: "contacts" })}</Link>
        </li>
      </Nav>
    </div>
  );
};
