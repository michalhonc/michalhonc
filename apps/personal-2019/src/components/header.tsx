import * as React from "react";
import { useStaticQuery, graphql } from "gatsby";
import BackgroundImage from "gatsby-background-image";
import { Link, useIntl, changeLocale } from "gatsby-plugin-intl";

import { Logo } from "./logo";
import Translate from "../images/translate.svg";
import { ReservationConsultation } from "./reservationConsultation";
import { Nav } from "./nav";

const Query = graphql`
  {
    background: file(relativePath: { eq: "header-background.jpg" }) {
      childImageSharp {
        fluid(maxWidth: 1920, quality: 90) {
          ...GatsbyImageSharpFluid_withWebp
        }
      }
    }
    foreground: file(relativePath: { eq: "header-foreground.png" }) {
      childImageSharp {
        fluid(maxWidth: 1920, quality: 100) {
          ...GatsbyImageSharpFluid_withWebp
        }
      }
    }
  }
`;

export const Header: React.FC = () => {
  const intl = useIntl();
  const data = useStaticQuery(Query);

  const { fluid: background } = data.background.childImageSharp;
  const { fluid: foreground } = data.foreground.childImageSharp;

  return (
    <header className="Header">
      <BackgroundImage
        Tag="div"
        className="Header-background"
        fluid={background}
      />
      <BackgroundImage
        Tag="div"
        className="Header-foreground"
        fluid={foreground}
      />

      <div className="Header-hero">
        <h1>
          <div>{intl.formatMessage({ id: "hero_1" })}</div>
          <div
            dangerouslySetInnerHTML={{
              __html: intl.formatMessage({ id: "hero_2" }),
            }}
          />
          <div>{intl.formatMessage({ id: "hero_3" })}</div>
          <div
            dangerouslySetInnerHTML={{
              __html: intl.formatMessage({ id: "hero_4" }),
            }}
          />
        </h1>
        <ReservationConsultation />
      </div>

      <Logo />

      <Nav canChangeLocal>
        <li>
          <Link to="/blog">{intl.formatMessage({ id: "blog" })}</Link>
        </li>
        <li>
          <a href="#kontakty">{intl.formatMessage({ id: "contacts" })}</a>
        </li>
      </Nav>

      {/* <button className="Header-button">
        <span><FormattedMessage id="download_CV" /></span>
      </button> */}
    </header>
  );
};
