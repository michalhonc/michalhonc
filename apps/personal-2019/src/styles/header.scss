$blur: 0.2rem;

@keyframes blur {
  0% {
    filter: blur(0);
  }
  100% {
    filter: blur($blur);
  }
}
@keyframes slideInFromRight {
  0% {
    transform: translateX(20rem);
  }
  100% {
    transform: translateX(0);
  }
}
@keyframes slideInFromLeft {
  0% {
    transform: translateX(-20rem);
  }
  100% {
    transform: translateX(0);
  }
}

.Header {
  position: relative;
  width: calc(100% + 2 * #{$blur});
  height: 100vh;
  min-height: 50rem;
  overflow: hidden;

  &-foreground,
  &-background {
    position: absolute;
    width: calc(100% + 2 * #{$blur});
    height: 101vh;

    @include respond('phone') {
      width: calc(100% + 15rem);
    }
  }

  &-background {
    animation: 0.5s ease-out 0s blur;
    animation-fill-mode: both;
    min-height: 50rem;
    top: -$blur - 0.2rem;
    left: -$blur;
    right: -$blur;
  }

  &-foreground {
    animation: 1s ease-out 0s slideInFromLeft;
    animation-fill-mode: both;
    margin-top: -100vh;
    top: 5rem;
  }

  &-navigation {
    position: absolute;
    top: 2rem;
    right: 15%;
    display: flex;
    text-transform: uppercase;

    @include respond('tab-port') {
      flex-direction: column;
      align-items: flex-end;
    }



    & li {
      &:not(:last-of-type) {
        margin-right: 3rem;
        @include respond('tab-port') {
          margin-right: 0;
          margin-bottom: 1rem;
        }
      }

      & a {
        color: #fff;
        font-size: 1.8rem;

        &:hover {
          color: #dedede;
        }
      }
    }

  }

  &-logo {
    height: 29.5rem;
    width: 40rem;
    position: absolute;
    top: 0;
    left: 5%;

    @include respond('phone') {
      left: -10%;
      width: 20rem;
      height: 19.5rem;
    }

    & > div {
      position: relative;
    }

    &-background {
      position: absolute;
      top: 0;

      @include respond('phone') {
        width: 20rem;
        height: 14.8rem;
      }
    }

    &-foreground {
      position: absolute;
      top: 3rem;
      left: 50%;
      transform: translateX(-50%);

      @include respond('phone') {
        top: 2rem;
        width: 11rem;
        height: 3.5rem;
      }

    }
  }

  &-changeLocale {
    position: absolute;
    height: 2.4rem;
    width: 2.4rem;
    top: 2rem;
    right: 5%;
  }

  &-hero {
    animation: 1s ease-out 0s slideInFromRight;
    animation-fill-mode: both;
    position: absolute;
    bottom: 40%;
    left: 50%;
    margin-right: 1.6rem;

    @include respond('phone') {
      left: 40%;
    }

    & .ReservationConsultation {
      margin-top: 5rem;

      @include respond('phone') {
        margin-top: 3rem;
      }
    }

    & h1 {
      & div {
        font-size: 2.4rem;
        color: #fff;

        @include respond('phone') {
          font-size: 1.4rem;
        }

        &:not(:last-of-type) {
          margin-bottom: 2rem;
          @include respond('phone') {
            margin-bottom: 0.5rem;
          }
        }

        &:nth-of-type(2) {
          font-size: 4rem;

          @include respond('phone') {
            font-size: 2.4rem;
          }
        }

        &:nth-of-type(4) {
          font-size: 3rem;

          @include respond('phone') {
            font-size: 1.8rem;
          }
        }
      }
    }
  }
}
