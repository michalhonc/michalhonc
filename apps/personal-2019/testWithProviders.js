import React from "react";
import renderer from "react-test-renderer";
import { IntlContextProvider } from "gatsby-plugin-intl";

export const renderWithProviders = (component) =>
  renderer.create(
    <IntlContextProvider value={{ language: "en-US", routed: true }}>
      {component}
    </IntlContextProvider>
  );
