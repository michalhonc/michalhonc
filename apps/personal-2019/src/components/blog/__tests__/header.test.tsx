import * as React from "react";
import renderer from "react-test-renderer";
import { renderWithProviders } from "../../../../testWithProviders";
import { Header } from "../header";

describe("[Blog] header", () => {
  it("renders correctly", () => {
    const tree = renderWithProviders(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
