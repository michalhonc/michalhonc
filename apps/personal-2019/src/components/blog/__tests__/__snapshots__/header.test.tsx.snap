// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Blog] header renders correctly 1`] = `
<div
  className="Blog-header"
>
  <div
    className="Header-logo"
  >
    <div>
      <test-file-stub
        className="Header-logo-background"
      />
      <a
        aria-label="Michal Honc Logo"
        href="/en-US/"
        onClick={[Function]}
      >
        <test-file-stub
          className="Header-logo-foreground"
        />
      </a>
    </div>
  </div>
  <nav>
    <ul
      className="Header-navigation"
    >
      <li>
        <a
          href="/en-US/"
          onClick={[Function]}
        >
          O mnÄ›
        </a>
      </li>
      <li>
        <a
          href="/en-US/blog"
          onClick={[Function]}
        >
          Blog
        </a>
      </li>
      <li>
        <a
          href="/en-US/#kontakty"
          onClick={[Function]}
        >
          Kontakty
        </a>
      </li>
    </ul>
    <button
      aria-label="Change language to english"
      className="Header-changeLocale"
      onClick={[Function]}
      type="button"
    >
      <test-file-stub />
    </button>
  </nav>
</div>
`;
